postgresql/resource_postgresql_grant.go

```diff
@@ -489,15 +631,44 @@ func createRevokeQuery(d *schema.ResourceData) string {
 			pq.QuoteIdentifier(srvName.(string)),
 			pq.QuoteIdentifier(d.Get("role").(string)),
 		)
-	case "TABLE", "SEQUENCE", "FUNCTION":
+	case "COLUMN":
 		objects := d.Get("objects").(*schema.Set)
-		if objects.Len() > 0 {
+		columns := d.Get("columns").(*schema.Set)
+		privileges := d.Get("privileges").(*schema.Set)
+
+		if privileges.Len() == 0 || columns.Len() == 0 {
+			// No privileges to revoke, so don't revoke anything
+			query = "SELECT NULL"
+		} else {
 			query = fmt.Sprintf(
-				"REVOKE ALL PRIVILEGES ON %s %s FROM %s",
-				strings.ToUpper(d.Get("object_type").(string)),
+				"REVOKE %s (%s) ON TABLE %s FROM %s",
```

Comment:
Hi @cyrilgdn 👋🏼 
Thanks for reading the PR!
The issue with revoking all privileges is that it will also revoke privileges that were created **outside** Terraform.
Doing so might negatively impact users of the provider and create weird bugs for them.

What do you think?

Reply:
One reason `REVOKE ALL PRIVILEGES` may be preferred: I've been bitten a few times where the column has been removed, so if I attempt to then remove the column grant via terraform, I get an error saying the column doesn't exist. An alternative would be to gracefully handle such an error.

Reply:
I support @kda-jt here.

I agree that we should **not** let the provider use `REVOKE ALL PRIVILEGES` as it may erase privileges defined outside of Terraform management (e.g. missed when porting over to Terraform), which could lead to unexpected & difficult to resolve application breakages. 

Only what has been created by Terraform should be removable by Terraform.

Regarding revoking permissions from deleted columns/tables - fine, it should either be handled, or either we accept that it might occur and that users might have to run `terraform state rm` and revoke it manually. It definitely does not warrant risking to revoke unmanaged but necessary privileges.

Reply:
@bismark the most graceful way to handle such an error would be to run `terraform state rm <offending_column_grant>`.

IMO, the provider should not handle column deletions for the following reasons:

:one: Terraform resources only manage the physical resources they "own". In this instance `postgresql_grant` does not own the column, it only owns and manages the grant **on** the column.
HashiCorp's provider design principles also state that [resources should represent a single API object.](https://www.terraform.io/plugin/hashicorp-provider-design-principles#resources-should-represent-a-single-api-object)

:two: "Gracefully handling such errors" would require the provider to be able to distinguish between a grant lost through column deletion, and a grant lost through a simple `REVOKE`. This entails managing the column's existence in the state file, which goes against :one:

> Furthermore, I firmly believe that a postgresql_grant resource creation should not be able to run any REVOKE statement (e.g. while being passed privileges=[]) as most of the times, individuals will operate on the basis that green resource creations in Terraform are not destructive and therefore not dangerous - especially if they are called "grant". Nobody would want to be in an outage situation having to explain that a green plan filled with "grant" resource creations revoked privileges because of some hidden magic under the hood of some open-source software supposed to help the firm.

I wholeheartedly agree with @Quentin-M on this point.
Postgre's permission system is additive. Having destructive grants would go against this principle, and would be counter-intuitive. 
Furthermore, HashiCorp's design principles say that [resources should be importable](https://www.terraform.io/plugin/hashicorp-provider-design-principles#resources-should-be-importable), and honestly I can't see how we would import the absence of a grant.